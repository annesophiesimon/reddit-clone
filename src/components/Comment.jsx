const Comment = (comment) => {
  const { name, time, content } = comment;
  return (
    <div className="bg-white shadow-lg p-5 m-2 rounded-xl md:w-5/6">
      <div>
        <p>{name}</p>
        <p> {time} hours ago</p>
      </div>
      <p> {content}</p>
    </div>
  );
};

export default Comment;
