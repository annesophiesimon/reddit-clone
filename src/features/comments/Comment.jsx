import { convertTime } from '../../helpers';
/* eslint-disable react/prop-types */
// add nested comments ?
const Comment = (props) => {
  const { comments } = props;
  return (
    <>
      {comments?.length > 0 &&
        comments.map((comment, index) => (
          <div key={index} className="bg-[#f2f2f2] shadow-lg p-3 m-5 rounded-lg">
            <div>
              <div className="flex justify-between">
                <h1 className="font-bold text-[#212427] text-sm"> {comment.author}</h1>
                <p className="text-slate-500 text-xs"> {convertTime(comment.created_utc)}</p>
              </div>
              <p className=" text-[#212427]"> {comment.body}</p>
            </div>
          </div>
        ))}
    </>
  );
};

export default Comment;
