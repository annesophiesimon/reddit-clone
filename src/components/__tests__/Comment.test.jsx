import Comment from '../Comment';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('Comment', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Comment />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(<Comment />);
    const textElement = screen.getByText('hours ago');

    expect(textElement).toBeInTheDocument();
  });

  it('renders name, time and content', () => {
    render(<Comment name="Lea" time={8} content="bla bla bla" />);
    const nameElement = screen.getByText('Lea');
    const timeElement = screen.getByText('8 hours ago');
    const contentElement = screen.getByText('bla bla bla');

    expect(nameElement).toBeInTheDocument();
    expect(timeElement).toBeInTheDocument();
    expect(contentElement).toBeInTheDocument();
  });
});
