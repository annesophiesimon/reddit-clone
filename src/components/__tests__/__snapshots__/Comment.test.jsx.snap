// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders correctly 1`] = `
<div
  className="bg-white shadow-lg p-5 m-2 rounded-xl md:w-5/6"
>
  <div>
    <p />
    <p>
       
       hours ago
    </p>
  </div>
  <p>
     
  </p>
</div>
`;
